logexp/cmd/main.go

```diff
@@ -0,0 +1,253 @@
+}
+
+// Check if the config provided through the flags take valid values.
+func checkConfigValidity() error {
+	glog.Infof("Verifying if a valid config has been provided through the flags")
+	if nodeName == "" {
+		return fmt.Errorf("Flag --node-name has its value unspecified")
+	}
+	if gcsArtifactsPath == "" {
+		return fmt.Errorf("Flag --gcs-artifacts-path has its value unspecified")
+	}
+	if _, err := os.Stat(gcloudAuthFilePath); err != nil {
+		return fmt.Errorf("Could not find the gcloud service account file: %v", err)
+	} else {
+		cmd := exec.Command("gcloud", "auth", "activate-service-account", "--key-file=" + gcloudAuthFilePath)
+		if err := cmd.Run(); err != nil {
+			return fmt.Errorf("Failed to activate gcloud service account: %v", err)
+		}
+	}
+	return nil
+}
+
+// Create logfile for systemd service in outputDir with the given journalctl outputMode.
+func createSystemdLogfile(service string, outputMode string, outputDir string) error {
+	// Generate the journalctl command.
+	var journalCmd string
+	if service == "kern" {
+		journalCmd = fmt.Sprintf("sudo journalctl --output=%v -k", outputMode)
+	} else {
+		journalCmd = fmt.Sprintf("sudo journalctl --output=%v -u %v.service", outputMode, service)
+	}
+	cmd := exec.Command("/bin/sh", "-c", journalCmd)
+
+	// Run the command and record the output to a file.
+	output, err := cmd.Output()
+	if err != nil {
+		return fmt.Errorf("Journalctl command for '%v' service failed: %v", service, err)
+	}
+	logfile := filepath.Join(outputDir, service + ".log")
+	if err := ioutil.WriteFile(logfile, output, 0444); err != nil {
+		return fmt.Errorf("Writing to file of journalctl logs for '%v' service failed: %v", service, err)
+	}
+	return nil
+}
+
+// Create logfiles for systemd services in outputDir, based on whether the node isMaster.
+func createSystemdLogfiles(outputDir string, isMaster bool) {
+	services := systemdServices
+	if ! isMaster {
+		services = append(services, nodeSystemdServices...)
+	}
+	for _, service := range services {
+		if err := createSystemdLogfile(service, "cat", outputDir); err != nil {
+			glog.Warningf("Failed to record journalctl logs: %v", err)
+		}
+	}
+	// Service logs specific to VM setup.
+	for _, service := range systemdSetupServices {
+		if err := createSystemdLogfile(service, "short-precise", outputDir); err != nil {
+			glog.Warningf("Failed to record journalctl logs: %v", err)
+		}
+	}
+}
+
+// Copy logfiles specific to this node based on its type, cloud-provider, system services, etc
+// to a temporary directory. Also create logfiles for systemd services if journalctl is present.
+// We do not expect this function to see an error.
+func prepareLogfiles(logDir string) {
+	glog.Infof("Preparing logfiles relevant to this node")
+	logfiles := make([]string, 0)
+	var isMaster bool
+
+	// Select node-type (master/node) specific logs.
+	if _, err := os.Stat(localLogPath + "/kube-apiserver.log"); err == nil {
```

Comment:
Removed it. Infact removed all the logging code specific to master as currently this doesn't anyway work for master (due to some gcloud auth issue).
And in fact the main chunk of the time we are trying to optimize is that due to copying of logs from nodes. Master logs copying can still continue to be done by scp'ing through jenkins.
Also, this way we would not have to complicate our existing setup for kubemark master too.